File: year2020/month12/day1210/ms_1210.cpp
#include <iostream>
#include <string>
#include <vector>
#include <sstream>

using namespace std;

vector<int> input();
int solution(vector<int> a, vector<int> b);

int main() {
	vector<int> a = input();
	vector<int> b = input();

	if ((1 <= a.size() && a.size() <= 1000) && (1 <= b.size() && b.size() <= 1000)) {
		if (a.size() == b.size()) {
			cout << solution(a, b) << endl;
		}
	}

	return 0;
}

vector<int> input() {
	string Input;
	vector<int> temp;

	getline(cin, Input);

	stringstream ss(Input);

	while (getline(ss, Input, ' ')) {
		int value = stoi(Input);

		if (-1000 <= value && value <= 1000) {
			temp.push_back(value);
		}
	}

	return temp;
}

int solution(vector<int> a, vector<int> b) {
	int answer = 0;
	int len = a.size();

	for (int i = 0; i < len; i++) {
		answer += a[i] * b[i];
	}

	return answer;
}

File: year2020/month12/day1210/ms_1210.cs
using System;

namespace algorithm_119
{
    class ms_1210
    {
        static int solution(int[] a, int[] b)
        {
            int answer = 0;
        
            for(int i = 0; i < a.Length; i++)
            {
                answer += a[i] * b[i];
            }
        
            return answer;
        }

        static void Main(string[] args)
        {
            String[] Input = Console.ReadLine().Split(' ');
            int i = 0;
            int[] A = new int[Input.Length];

            foreach(var data in Input)
            {
                int temp = Convert.ToInt32(data);

                if(-1000 <= temp && temp <= 1000)
                {
                    A[i] = Convert.ToInt32(data);
                }

                i++;
            }

            Input = Console.ReadLine().Split(' ');
            i = 0;
            int[] B = new int[Input.Length];

            foreach(var data in Input)
            {
                int temp = Convert.ToInt32(data);

                if(-1000 <= temp && temp <= 1000)
                {
                    B[i] = Convert.ToInt32(data);
                }
                
                i++;
            }

            if((1 <= A.Length && A.Length <= 1000) && (1 <= B.Length && B.Length <= 1000))
            {
                if(A.Length == B.Length) 
                {
                    Console.WriteLine(solution(A, B));
                }
            }
        }
    }
}
